pybamm/discretisations/discretisation.py

```diff
@@ -155,18 +155,13 @@ def process_dict(self, var_eqn_dict):
 
         """
 
-        for eqn_key, eqn in var_eqn_dict.items():
-
+        for var, eqn in var_eqn_dict.items():
+            # Broadcast if the equation evaluates to a number(e.g. Scalar)
             if eqn.evaluates_to_number():
-                if eqn_key.domain == []:
-                    eqn = pybamm.NumpyBroadcast(eqn, eqn_key.domain, {})
-                else:
-                    eqn = self._spatial_methods[eqn_key.domain[0]].broadcast(
-                        eqn, eqn_key.domain
-                    )
+                eqn = pybamm.Broadcast(eqn, var.domain)
 
             # Process symbol (original or broadcasted)
-            var_eqn_dict[eqn_key] = self.process_symbol(eqn)
+            var_eqn_dict[var] = self.process_symbol(eqn)
```

Comment:
might be just because pull request was made before previous merges but should this still read eqn.id?